minicopier-master/src/minicopier/Configuration.java

```diff
@@ -0,0 +1,187 @@
+/*
+  Configuration.java / MiniCopier
+  Copyright (C) 2007-2009 Adrian Courrèges
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License as
+  published by the Free Software Foundation; either version 2 of
+  the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+*/
+
+package minicopier;
+
+import java.util.*;
+import java.io.*;
+
+import minicopier.gui.DialogMsg;
+
+public class Configuration {
+	
+	private static final String configName = "configuration.ini";
+	
+	public static String language;
+	
+	public static boolean alwaysOnTop;
+	
+	//Look And Feel
+	//"pgs" (default)
+	//"native"
+	//"metal"
+	public static String lookAndFeel;
+	
+	//What to do if destination file already exists
+	// 0 -> ask user
+	// 1 -> ignore
+	// 2 -> overwrite
+	// 3 -> overwrite if older
+	// 4 -> resume
+	// 5 -> rename
+	public static int collisionAction;
+	
+	//What to do if a source path contains a symbolic link
+	//(Unix systems only)
+	// 0 -> follow
+	// 1 -> ignore
+	public static int symbolicLinkAction;
+    
+    //Is speed limit active?
+    public static boolean limitSpeed;
+    
+    //Maximum throughput of the transfer
+    public static int nbBytesPerS;
+    
+    private static boolean firstLaunch = false;
+	
+	public Configuration(){
+        
+        this.limitSpeed = false;
+        
+        this.nbBytesPerS = 1024*1024;
+		
+		Properties prop = new Properties(); 
+		
+		File configFile = new File(configName);
+		
+		if (!configFile.exists()) {
+			System.out.println("Configuration file not found. Creating configuration.ini.");
+			createConfigFile();
+			firstLaunch = true;
+		}
+		
+		try {
+			FileInputStream in = new FileInputStream(configName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *in*. The resource is referenced in statements at the following line: 82. The resource closure statement is at line: 83. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by Properties.load. Either a) close *in* in a try-finally block or b) close the resource by declaring *in* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).